src/apps/ipv4/arp.lua

```diff
@@ -20,16 +20,17 @@ local datagram = require("lib.protocol.datagram")
 local ethernet = require("lib.protocol.ethernet")
 local ipv4     = require("lib.protocol.ipv4")
 local alarms = require("lib.yang.alarms")
+local S = require("syscall")
 
 alarms.add_to_inventory {
   [{alarm_type_id='arp-resolution'}] = {
-    resource={'nic-v4'},
+    resource=tostring(S.getpid()),
```

Comment:
This code will run in the main process and in the worker processes, and so it will get two different resource identifiers.  Is that what you want?

Reply:
OK offline you explained that the idea is that since the worker has to generate a fresh resource id, that the idea is that the worker tells the main process about the set of alarms that it might raise.  Given that that's the case, we can actually move alarm declaration per-app, and not global, because each app can have its own resource id's.  Something to fix in a followup but after our release I think.